rest/config_legacy_test.go

```diff
@@ -71,6 +71,18 @@ func TestLegacyConfigToStartupConfig(t *testing.T) {
 			input:    LegacyServerConfig{},
 			expected: StartupConfig{API: APIConfig{AdminInterfaceAuthentication: base.BoolPtr(true)}},
 		},
+		{
+			name:     "http:// to couchbase://",
+			base:     StartupConfig{},
+			input:    LegacyServerConfig{Databases: DbConfigMap{"db": &DbConfig{BucketConfig: BucketConfig{Server: base.StringPtr("http://http.couchbase.com:2929,host2,host1:22,[2001:db8::8811],[2001:db8::8822]:888")}}}},
+			expected: StartupConfig{Bootstrap: BootstrapConfig{Server: "couchbase://http.couchbase.com,host2,host1,[2001:db8::8811],[2001:db8::8822]"}},
```

Comment:
Plenty of combinations across these 2 tests which is good to see, although I think we should also be covering these separate cases, to ensure we're not changing already-compatible connection strings.

- `http://foo,bar?foo=fooval&bar=barval` -> `couchbase://foo,bar?foo=fooval&bar=barval`
  - ensure all query parameters are preserved in the new URL
- `couchbase://foo:21210` -> `couchbase://foo:21210`
  - ensure custom ports for `couchbase(s)://` schemes are retained